cmake/SimbodyConfig.cmake.in

```diff
@@ -70,152 +63,13 @@ if (NOT "@SIMBODY_DOXYGEN_TAGFILE_NAME@" STREQUAL "")
-      NO_DEFAULT_PATH)
-    if(NOT onelibd)
-      message(FATAL_ERROR 
-          "Library '${lib}' in package Simbody is not installed properly")
-    endif()
-    set(LIBS ${LIBS} "${onelibd}")
-  endforeach()
-endif()
-
-if (LIBS)
-    set(LIBS ${LIBS} "@MATH_LIBS_TO_USE@")
-    set(Simbody_LIBRARIES ${LIBS} CACHE STRING 
-        "Simbody dynamic libraries" FORCE)
-else()
-    set(Simbody_LIBRARIES Simbody_LIBRARIES-NOTFOUND CACHE STRING 
-        "Simbody dynamic libraries" FORCE)
-endif()
-
-# Static library
-set(LIBS)
-if(Simbody_STATIC_LIBRARY AND Simbody_STATIC_DEBUG_LIBRARY)
-  foreach(lib @PKG_LIBRARIES@)
-    set(lib "${lib}_static")
-    set(onelib  "${lib}-NOTFOUND"   CACHE INTERNAL "nothing")
-    set(onelibd "${lib}@CMAKE_DEBUG_POSTFIX@-NOTFOUND" CACHE INTERNAL "nothing")
-    find_library(onelib ${lib}
-      PATHS ${Simbody_LIB_DIR}
-      NO_DEFAULT_PATH)
-    find_library(onelibd ${lib}@CMAKE_DEBUG_POSTFIX@
-      PATHS ${Simbody_LIB_DIR}
-      NO_DEFAULT_PATH)
-    if((NOT onelib) OR (NOT onelibd))
-      message(FATAL_ERROR 
-          "Library '${lib}' in package Simbody is not installed properly")
-    endif()
-    set(LIBS ${LIBS} optimized "${onelib}" debug "${onelibd}")
-  endforeach()
-elseif(Simbody_STATIC_LIBRARY)
-  foreach(lib @PKG_LIBRARIES@)
-    set(lib "${lib}_static")
-    set(onelib  "${lib}-NOTFOUND"   CACHE INTERNAL "nothing")
-    find_library(onelib ${lib}
-      PATHS ${Simbody_LIB_DIR}
-      NO_DEFAULT_PATH
-      )
-    if(NOT onelib)
-      message(FATAL_ERROR 
-          "Library '${lib}' in package Simbody is not installed properly")
-    endif()
-    set(LIBS ${LIBS} "${onelib}")
-  endforeach()
-elseif(Simbody_STATIC_DEBUG_LIBRARY)
-  foreach(lib @PKG_LIBRARIES@)
-    set(lib "${lib}_static@CMAKE_DEBUG_POSTFIX@")
-    set(onelibd  "${lib}-NOTFOUND"   CACHE INTERNAL "nothing")
-    find_library(onelibd ${lib}
-      PATHS ${Simbody_LIB_DIR}
-      NO_DEFAULT_PATH
-      )
-    if(NOT onelibd)
-      message(FATAL_ERROR 
-          "Library '${lib}' in package Simbody is not installed properly")
-    endif()
-    set(LIBS ${LIBS} "${onelibd}")
-  endforeach()
-endif()
-
-if (LIBS)
-    # these aren't available in static
-    set(LIBS ${LIBS} "@MATH_LIBS_TO_USE@")
-    set(Simbody_STATIC_LIBRARIES "${LIBS}" CACHE STRING 
-        "Simbody static libraries" FORCE)
-else()
-    set(Simbody_STATIC_LIBRARIES Simbody_STATIC_LIBRARIES-NOTFOUND CACHE STRING 
-        "Simbody static libraries" FORCE)
-endif()
-
-# This CMake-supplied script provides standard error handling.
-include(FindPackageHandleStandardArgs OPTIONAL)
-find_package_handle_standard_args(Simbody DEFAULT_MSG 
-    Simbody_INCLUDE_DIR)
+# These are IMPORTED targets created by SimbodyTargets.cmake
+set(Simbody_LIBRARIES @SimTKCOMMON_SHARED_LIBRARY@ @SimTKMATH_SHARED_LIBRARY@ @SimTKSIMBODY_SHARED_LIBRARY@)
+set(Simbody_STATIC_LIBRARIES @SimTKCOMMON_STATIC_LIBRARY@ @SimTKMATH_STATIC_LIBRARY@ @SimTKSIMBODY_STATIC_LIBRARY@)
```

Comment:
I had wrongly assumed that variables like `SimTKCOMMON_STATIC_LIBRARY` would only be defined if `BUILD_STATIC_LIBRARY` was ON, but this is not the case. I can (change all the CMake files or ) wrap the `set(Simbody_STATIC_LIBRARIES ...` line in a `if(@BUILD_STATIC_LIBRARIES@)` block. I don't think you can disable compilation of shared libraries.

Anyway, this is a really tricky point: how is client code supposed to deal with the un/availability of static libraries? Does it just check if `Simbody_STATIC_LIBRARIES` is defined? I will try to see if there is a way for client code to specify which libraries they want in the `find_package` call